File: clients/csharp/DebugInterface.cs
using System.IO;

namespace Komus24
{
    public class DebugInterface
    {
        private BinaryWriter writer;
        private BinaryReader reader;
        public DebugInterface(BinaryReader reader, BinaryWriter writer)
        {
            this.reader = reader;
            this.writer = writer;
        }

        public void AddCircle(Komus24.Model.Vec2Double pos, double radius, Komus24.Debugging.Color color)
        {
            Add(new Komus24.Debugging.DebugData.Circle(pos, radius, color));
        }
        
        public void AddLine(Komus24.Model.Vec2Double point1, Komus24.Model.Vec2Double point2, double width, Komus24.Debugging.Color color)
        {
            Add(new Komus24.Debugging.DebugData.Line(point1, point2, width, color));
        }
        
        public void AddRect(Komus24.Model.Vec2Double corner1, Komus24.Model.Vec2Double corner2, Komus24.Debugging.Color color)
        {
            Add(new Komus24.Debugging.DebugData.Rect(corner1, corner2, color));
        }
        
        public void AddText(string text, Komus24.Model.Vec2Double pos, double size, double align, Komus24.Debugging.Color color)
        {
            Add(new Komus24.Debugging.DebugData.Text(text, pos, size, align, color));
        }

        public void Add(Komus24.Debugging.DebugData debugData)
        {
            Send(new Komus24.Debugging.DebugCommand.Add(debugData));
        }
        
        public void Clear()
        {
            Send(new Komus24.Debugging.DebugCommand.Clear());
        }
        
        public void SetAutoFlush(bool enable)
        {
            Send(new Komus24.Debugging.DebugCommand.SetAutoFlush(enable));
        }
        
        public void Flush()
        {
            Send(new Komus24.Debugging.DebugCommand.Flush());
        }

        public void Send(Komus24.Debugging.DebugCommand command)
        {
            new Komus24.Codegame.ClientMessage.DebugMessage(command).WriteTo(writer);
            writer.Flush();
        }

        public Komus24.Debugging.DebugState GetState()
        {
            new Komus24.Codegame.ClientMessage.RequestDebugState().WriteTo(writer);
            writer.Flush();
            return Komus24.Debugging.DebugState.ReadFrom(reader);
        }
    }
}

File: clients/python/debug_interface.py
from codegame.client_message import ClientMessage
from debugging.debug_command import DebugCommand
from debugging.debug_state import DebugState
from debugging.debug_data import DebugData
from stream_wrapper import StreamWrapper
from debugging.color import Color
from model.vec2_double import Vec2Double
from stream_wrapper import StreamWrapper

class DebugInterface:
    def __init__(self, reader, writer):
        self.reader = reader
        self.writer = writer
        
    def add_circle(self, pos: Vec2Double, radius: float, color: Color):
        self.add(DebugData.Circle(pos, radius, color))
    
    def add_line(self, point1: Vec2Double, point2: Vec2Double, width: float, color: Color):
        self.add(DebugData.Line(point1, point2, width, color))
    
    def add_rect(self, corner1: Vec2Double, corner2: Vec2Double, color: Color):
        self.add(DebugData.Rect(corner1, corner2, color))
    
    def add_text(self, text: str, pos: Vec2Double, size: float, align: float, color: Color):
        self.add(DebugData.Text(text, pos, size, align, color))

    def add(self, debug_data: DebugData):
        self.send(DebugCommand.Add(debug_data))
    
    def clear(self):
        self.send(DebugCommand.Clear())
    
    def set_auto_flush(self, enable: bool):
        self.send(DebugCommand.SetAutoFlush(enable))
    
    def flush(self):
        self.send(DebugCommand.Flush())

    def send(self, command: DebugCommand):
        ClientMessage.DebugMessage(command).write_to(self.writer)
        self.writer.flush()

    def get_state(self) -> DebugState:
        ClientMessage.RequestDebugState().write_to(self.writer)
        self.writer.flush()
        return DebugState.read_from(self.reader)